index.bs

```diff
@@ -535,101 +539,233 @@ dictionary WriteParams {
-  : await |stream| . {{FileSystemWritableFileStream/write()|write}}(|data|)
-  :: Writes the content of |data| into the file associated with |stream| at the current file
-     cursor offset in bytes from the top of the file by default.
+A {{FileSystemWritableFileStream}} object is a {{WritableStream}} object with additional
+convenience methods, which operates on a single file on disk.
 
-     No changes are written to the actual file until on disk until {{FileSystemWritableFileStream/close()}}
-     is called. Changes are typically written to a temporary file instead.
+Upon creation, an underlying sink will have been created and the stream will be usable.
+All operations executed on the stream are queuable and producers will be able to respond to backpressure.
 
-     If |data| is of type {{WriteParams}}, validates and executes the content of |data|, which might result in changes into the file associated with |stream|.
+The underlying sink's write method, and therefore {{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}}
+method, will accept byte-like data or {{WriteParams}} as input.
 
-     {{WriteParams}} requires the attribute {{WriteParams/type|type}} which determines what the operations to
-     execute are.
+The {{FileSystemWritableFileStream}} has a file position cursor initialized at byte offset 0 from the top of the file.
+When using {{FileSystemWritableFileStream/write()|write()}} or by using WritableStream capabilities through the {{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}} method, this position will be advanced based on the number of bytes written through the stream object.
 
-     If {{WriteParams/type|type}} is {{truncate}}, {{WriteParams/size|size}} is a <span class=allow-2119>required</span> attribute.
+Similarly, when piping a {{ReadableStream}} into a {{FileSystemWritableFileStream}} object, this position is updated with the number of bytes that passed through the stream.
 
-     If {{WriteParams/type|type}} is {{seek}}, {{WriteParams/position|position}} is a <span class=allow-2119>required</span> attribute.
+{{WritableStream/getWriter()|getWriter()}} returns an instance of {{WritableStreamDefaultWriter}}.
+</div>
 
-     If {{WriteParams/type|type}} is {{write}}, {{WriteParams/position|position}} is an <span class=allow-2119>optional</span> attribute, {{WriteParams/data|data}} is a <span class=allow-2119>required</span> attribute.
+<div algorithm>
+To <dfn>create a new FileSystemWritableFileStream</dfn> given a [=file entry=] `file`, perform the following steps:
+
+1. Let |stream| be a new {{FileSystemWritableFileStream}}.
+1. Perform [$InitializeWritableStream$](|stream|)
+1. Set |stream|.[=FileSystemWritableFileStream/[[file]]=] to [=file entry=].
+1. Let |controller| be a new {{WritableStreamDefaultController}}.
+1. Let |startAlgorithm| be an algorithm that returns `undefined`.
+1. Let |writeAlgorithm| be an algorithm which takes a |chunk| argument
```

Comment:
|writeAlgorithm| has to return a promise. It's fine to add "and returns a promise resolved with undefined." here.